content-api/content-actors/src/main/scala/org/sunbird/channel/managers/ChannelManager.scala

```diff
@@ -0,0 +1,88 @@
+package org.sunbird.channel.managers
+
+import java.util
+import java.util.concurrent.CompletionException
+import java.util.Optional
+
+import org.sunbird.common.dto.{Request, Response, ResponseHandler}
+import org.sunbird.util.ChannelConstants
+import org.apache.commons.lang3.StringUtils
+import org.sunbird.cache.impl.RedisCache
+import org.sunbird.common.exception.{ClientException, ResourceNotFoundException, ResponseCode, ServerException}
+import org.sunbird.common.{Platform, dto}
+import com.mashape.unirest.http.HttpResponse
+import com.mashape.unirest.http.Unirest
+import org.apache.commons.collections4.CollectionUtils
+import org.sunbird.common.JsonUtils
+import org.sunbird.graph.utils.ScalaJsonUtils
+
+import scala.collection.JavaConversions.mapAsJavaMap
+import scala.concurrent.{ExecutionContext, Future}
+import scala.collection.JavaConverters._
+
+object ChannelManager {
+
+  def validateLicense(request: Request)(implicit ec: ExecutionContext): Response = {
+    val response: Response = ResponseHandler.OK
+    if (StringUtils.isNotEmpty(request.getRequest.get(ChannelConstants.DEFAULT_LICENSE).asInstanceOf[String])) {
+      val licenseList = RedisCache.getList(ChannelConstants.LICENSE_REDIS_KEY)
+      if (!licenseList.isEmpty && licenseList.contains(request.getRequest.get(ChannelConstants.DEFAULT_LICENSE)))
+        response
+      else {
+        val licenseList = searchLicenseList()
+        if (licenseList.contains(request.getRequest.get(ChannelConstants.DEFAULT_LICENSE))) {
+          RedisCache.saveList(ChannelConstants.LICENSE_REDIS_KEY, licenseList)
+          response
+        } else
+          throw new ResourceNotFoundException("LICENSE_NOT_FOUND", "License " + request.getRequest.get(ChannelConstants.DEFAULT_LICENSE).asInstanceOf[String] + " does not exist")
+      }
+    } else
+      response
```

Comment:
We should use "property as relation" behaviour to implement this. No need to explicitly validate it.